src/geometry/types.rs

```diff
@@ -230,4 +314,26 @@ mod tests {
         // given, when, then
         assert_eq!(Vector::zero().normalized(), None)
     }
+
+    #[test]
+    fn rect_quadrant() {
+        // given
+        let mut rect = Rect { origin: Point { x: 0.0, y: 0.0 }, size: Size { width: 6.0, height: 8.0 } };
+
+        // when
+        let result = rect.quadrants();
+
+        let (rect1, rect2, rect3, rect4) = result;
```

Comment:
Like above, give these better names. Also you don't need to store it in result, you can destructure the return value of `quadrants()` directly.